### ISSUE TITLE
Tree component is not fully accessible

### ISSUE THREAD

**Issue by Mikalai-Povarau** [ant-design/ant-design#32844]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/i2wsm)

### Steps to reproduce
1. Enable voiceOver or other screenreader
2. Open the code sandbox link
3. Navigate to Tree using Tab
4. Hold Tab and using arrow keys navigate through the tree (left/right arrows to collapse/expand)

 * Expected Result
    Screen reader reads content of node, each tree node should be focusable and have abitily to be navigated via Tab / Shift + Tab .

 * Actual Result
    Screen reader doesn't read content of node. Focus is on the place as it was before, but we still can navigate through the tree.

### What is expected?
Each tree node should be focusable and have abitily to be navigated via Tab / Shift + Tab .
Each checkbox should be real checkbox with appropriate input type and label.
Clicking on checkbox label should check/uncheck the checkbox.
Solution should pass all requirements of WCAG 2.1, notably Success Criterion 3.2.2

https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html .

Please see an examle of accessible component made by w3:
https://www.w3schools.com/tags/att_input_type_checkbox.asp

### What is actually happening?
Tree component has several accessibility issues and not clearly understandable for users through assistive technolgies. 
Each of tree nodes are not focusable.
Clicking on checkbox label doesn't check/uncheck the checkbox.

| Environment | Info |
|---|---|
| antd | 4.16.13 |
| React | 16.14.0 |
| System | MacOS Big Sur / Windows 10 |
| Browser | Chrome 94.0.4606.81 / Safari 14.1.2 (only on MacOS) / Firefox 94.0.1 |

**Comment by afc163** [comment]
Duplicate of #31012
